### PR TITLE
Drop managed fields from objects

### PR DESCRIPTION
**Pull request by charlieegan3**
This adds weight to the agent uploads that we don't need.

Also includes changes from a go mod tidy.

Signed-off-by: Charlie Egan <charlieegan3@users.noreply.github.com>

### CODE REVIEW

File: pkg/datagatherer/k8s/dynamic.go [hunk 1]
```diff
@@ -211,12 +213,25 @@ func redactList(list *unstructured.UnstructuredList) error {
...
+			metadata, metadataPresent := object.Object["metadata"].(map[string]interface{})
+			if metadataPresent {
+				// Drop managed fields if set
+				if _, present := metadata["managedFields"]; present {
+					delete(metadata, "managedFields")
+				}
+			}
+
+			// update the metadata after an modifications
+			object.Object["metadata"] = metadata
```
Comment (tfadeyi) [inline pkg/datagatherer/k8s/dynamic.go:234]: Is this needed?

Comment (charlieegan3) [reply pkg/datagatherer/k8s/dynamic.go]: Huh, we don't. I just copied it from above. Removed in 2ba21c0092016487bce2a1bdba2525e60291cd5b

**Comment by charlieegan3** [comment]
Closed in favor of https://github.com/jetstack/preflight/pull/203

**Pull request closed without merging**